lib/linter/apply-disable-directives.js

```diff
@@ -18,6 +18,121 @@ function compareLocations(itemA, itemB) {
+ * @param {Directive[]} directives Unused directives to be removed.
+ * @param {ParentComment} parentComment Location and rule IDs for the comment to delete.
+ * @param {SourceCode} sourceCode A SourceCode object for the given text.
+ * @returns {{ fix, position, ruleIds }[]} Details for later creation of output Problems.
+ */
+function createIndividualDirectivesRemoval(directives, parentComment, sourceCode) {
+    const commentStart = sourceCode.lineStartIndices[parentComment.loc.start.line - 1] + parentComment.loc.start.column;
+    const commentToken = sourceCode.getTokenByRangeStart(commentStart, { includeComments: true });
+    const commentContents = commentToken.value.split("--")[0].trimRight();
+    let lastRuleEnd = 0;
+
+    return directives.map(directive => {
+        const ruleStart = commentContents.indexOf(directive.ruleId, lastRuleEnd);
+        const ruleEnd = lastRuleEnd = ruleStart + directive.ruleId.length;
+
+        return {
+            description: directive.ruleId,
+            fix: {
+                range: [
+                    commentStart + 2 + ruleStart,
+                    commentStart + 2 + ruleEnd + (commentContents[ruleEnd] === "," ? 1 : 0)
+                ],
+                text: ""
+            },
+            position: directive.unprocessedDirective
+        };
+    });
+}
+
+/**
+ * Creates a description of deleting an entire unused disable comment.
+ * @param {Directive[]} directives Unused directives to be removed.
+ * @param {ParentComment} parentComment Location and rule IDs for the comment to delete.
+ * @param {SourceCode} sourceCode A SourceCode object for the given text.
+ * @returns {{ fix, position, ruleIds }} Details for later creation of an output Problem.
+ */
+function createCommentRemoval(directives, parentComment, sourceCode) {
+    const range = [
+        sourceCode.lineStartIndices[parentComment.loc.start.line - 1] + parentComment.loc.start.column,
+        sourceCode.lineStartIndices[parentComment.loc.end.line - 1] + parentComment.loc.end.column
+    ];
+    const ruleIds = directives.filter(directive => directive.ruleId).map(directive => `'${directive.ruleId}'`);
+
+    return {
+        description: ruleIds.length <= 2
+            ? ruleIds.join(" or ")
+            : `${ruleIds.slice(0, ruleIds.length - 2).join(", ")}, or ${ruleIds[ruleIds.length - 1]}`,
+        fix: {
+            range,
+            text: " "
+        },
+        position: directives[0].unprocessedDirective
+    };
+}
+
+/**
+ * Parses details from directives to create output Problems.
+ * @param {Directive[]} allDirectives Unused directives to be removed.
+ * @param {SourceCode} sourceCode A SourceCode object for the given text.
+ * @returns {{ fix, position, ruleIds }} Details for later creation of output Problems.
+ */
+function processUnusedDisableDirectives(allDirectives, sourceCode) {
+    const directiveGroups = groupByLine(allDirectives);
+
+    return lodash.flatMap(directiveGroups).map(directives => {
```

Comment:
lodash has been removed from deps in https://github.com/eslint/eslint/pull/14287. can you just use the `Array.flatMap()`?

Reply:
Exciting! `Array.flatMap()` is unavailable until support for Node 10 is dropped, but I can use the `flatMap` added in that PR once I merge in from `eslint:master`. 👍